engine/src/main/java/org/terasology/logic/config/ModuleConfigSystem.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2015 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.logic.config;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.registry.In;
+import org.terasology.registry.Share;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+@RegisterSystem
+@Share(ModuleConfigManager.class)
+public class ModuleConfigSystem extends BaseComponentSystem implements ModuleConfigManager {
+    private static final Logger logger = LoggerFactory.getLogger(ModuleConfigSystem.class);
+
+    @In
+    private EntityManager entityManager;
+
+    private Map<String, Map<String, String>> propertiesPerModule = new HashMap<>();
+
+    @Override
+    public void preBegin() {
+        for (EntityRef entityRef : entityManager.getEntitiesWith(ModuleConfigComponent.class)) {
```

Comment:
As this only happens once and doesnt get any changes that an entity gets,  would it be better suited to find all prefabs with ModuleConfigComponent?  
Now that I have tried it, how do you create the entities at all before preBegin? I tried the AutoCreateComponent,  but that didnt work.  I must be missing something.
Using prefabs instead of entities made it work well for my little test.  Could be a good way going forward.